File: src/main/kotlin/wowsft/model/gameparams/ship/component/flightcontrol/FlightControl.java
package wowsft.model.gameparams.ship.component.flightcontrol;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.util.LinkedHashMap;
import java.util.List;

@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class FlightControl
{
    private LinkedHashMap<String, Integer> consumablePlanesReserve;
    private List<Object> consumableSquadrons;
    private boolean hasConsumablePlanes;
    @JsonProperty("isGroundAttackEnabled")
    private boolean groundAttackEnabled;
    private LinkedHashMap<String, Integer> planesReserveAssignment;
    private float prepareTimeFactor;
    private List<String> squadrons;
}

File: src/main/kotlin/wowsft/model/gameparams/ship/component/flightcontrol/FlightControl.kt
package wowsft.model.gameparams.ship.component.flightcontrol

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonAlias

import java.util.LinkedHashMap

@JsonIgnoreProperties(ignoreUnknown = true)
class FlightControl {
    var consumablePlanesReserve = LinkedHashMap<String, Int>()
    var consumableSquadrons = ArrayList<Any>()
    var hasConsumablePlanes = false
    @JsonAlias("isGroundAttackEnabled")
    var groundAttackEnabled = false
    var planesReserveAssignment = LinkedHashMap<String, Int>()
    var prepareTimeFactor = 0.0
    var squadrons = ArrayList<String>()
}
